### DISCUSSION TITLE
Can we make tags in TagsInput editable ?

### DISCUSSION THREAD

**Discussion by carezkh** [mantinedev/mantine#5519] [Feature requests]
It seems the tags in TagsInput component is not editable currently, i have to remove it and then input the new one, it's inconvenient for the tags with long string.  Or can we copy-and-paste the string to TagsInput in editable plain text, and convert it to tag only when `Enter` is pressed ?
Thanks !

**Comment by rtivital** [comment]
It is not planned to add this feature to the library, you can build a custom component for that with Combobox – https://mantine.dev/core/combobox/

**Reply by carezkh** [reply]
Is there any method/workaround to customize the existed TagsInput component in the lib? Build a totally new one is not easy for me, although the goals can achieved by changing the source code in the lib.

**Reply by mrclrchtr** [reply]
I have the same problem. I would like to be able to easily change the background color of pills, depending on their state. Is there any easy way to do this without having to build a new one?

**Reply by rtivital** [reply]
No, this feature is not supported by TagsInput component. You can build a custom component with Combobox.

**Reply by mrclrchtr** [reply]
@rtivital I have customized PillsInput so that you can pass a map with “Value=Style” in order to assign a style to the pills based on a value. Would you be interested in a PR to incorporate this directly into Mantine?

In short, it looks something like this:
```
<StyleableTagsInput
        ...
        value={props.containerStates.map((it) => it.containerName)}
        styledValues={mapToStyledValues(props.containerStates)}
        ...
      />
```
```
function mapToStyledValues(containerStates: ContainerState[]): Map<string, Styles<PillFactory>> {
    const map = new Map<string, Styles<PillFactory>>();
    containerStates.forEach((containerState) => {
      let color;
      switch (containerState.state) {
        case "empty":
          color = "var(--mantine-color-green-light)";
          break;
        case "non-existent":
          color = "var(--mantine-color-blue-light)";
          break;
        case "filled":
          color = "var(--mantine-color-red-light)";
          break;
        default:
          color = "var(--mantine-color-grey-light)";
          break;
      }
      map.set(containerState.containerName, {
        root: { backgroundColor: color },
      });
    });
    return map;
  }
```
```
const values = _value.map((item, index) => {
    const styles = styledValues?.get(item);

    return (
      <Pill
        key={`${item}-${index}`}
        withRemoveButton={!readOnly}
        onRemove={() => setValue(_value.filter((i) => item !== i))}
        unstyled={false}
        styles={styles}
        className={getStyles("pill").className}
      >
        {item}
      </Pill>
    );
  });
```

**Reply by rtivital** [reply]
No, it is not planned to be added to the library.
